src-srv/utils/CollaborationServer.ts

```diff
@@ -231,6 +219,54 @@ export class CollaborationServer {
     return Y.encodeStateAsUpdate(yDoc)
   }
 
+  async #snapshotDocument({ documentName, document: yDoc, context }: onStoreDocumentPayload): Promise<void> {
+    // Disregard __inProgress documents
+    if ((yDoc.getMap('ele')
+      .get('root') as Y.Map<unknown>)
+      .get('__inProgress') as boolean) {
+      console.debug('::: saveDoc: Document is in progress, not saving')
+      return
+    }
+
+    // Convert yDoc to newsDoc, so we can hash it and maybe save it to the repository
+    const { document } = yDocToNewsDoc(yDoc)
+
+    const currentHash = createHash(JSON.stringify(document))
+
+    // Compare original hash with current hash to establish if there are any changes
+    // This solution relies on the same order of keys in the documents, but a change of key order
+    // should be indicate a change in the document anyway.
+    if (yDoc.getMap('hash').get('hash') === currentHash) {
+      console.debug('::: saveDoc: No changes in document')
+      return
+    }
+
+
+    const versionMap = yDoc.getMap('version')
+    const version = BigInt(versionMap.get('version') as string)
+
+    if (!document) {
+      throw new Error('No document to save')
+    }
+
+    const result = await this.#repository.saveDoc(document, context.token as string, version)
+    if (result?.status.code === 'OK') {
+      const connection = await this.#server.openDirectConnection(documentName, {
+        ...context,
+        agent: 'server'
+      })
+
+      await connection.transact(doc => {
+        const versionMap = doc.getMap('version')
+        const hashMap = doc.getMap('hash')
+        versionMap.set('version', result?.response.version.toString())
+        hashMap.set('hash', currentHash)
+      })
+
+      console.debug('::: Snapshot saved: ', result.response.version, 'new hash:', currentHash)
```

Comment:
Is this intentionally left in the code? Will produce a lot of log messages later on.